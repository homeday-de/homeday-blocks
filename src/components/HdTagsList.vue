<template>
  <section class="tags-list">
    <div
      v-for="item in items"
      :key="item"
      class="tags-list__tag">
      {{ item }}
    </div>
  </section>
</template>

<script>
export default {
  name: 'HdTagsList',
  props: {
    // Array of strings
    items: {
      type: Array,
      default: () => [],
    },
  },
};
</script>

<style lang="scss" scoped>
@import 'hd-blocks/styles/mixins.scss';

.tags-list {
  display: flex;
  flex-wrap: wrap;
  min-height: #{$stack-l + $stack-s};
  margin-bottom: $stack-m;
  &__tag {
    display: flex;
    align-items: center;
    height: $stack-l;
    background-color: rgba($heather, 0.5);
    border-radius: 3px;
    padding: 0 $inline-s;
    margin: $stack-s $inline-s 0 0;
  }
}
</style>
