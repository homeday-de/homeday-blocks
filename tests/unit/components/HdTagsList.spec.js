import { wrapperFactoryBuilder } from 'tests/unit/helpers';
import HdTagsList from '@/components/HdTagsList.vue';

const ITEMS = [
  'foo',
  'bar',
];

const wrapperBuilder = wrapperFactoryBuilder(HdTagsList, {
  props: {
    items: ITEMS,
  },
});

describe('HdTagsList', () => {
  it('renders component', () => {
    const wrapper = wrapperBuilder();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
