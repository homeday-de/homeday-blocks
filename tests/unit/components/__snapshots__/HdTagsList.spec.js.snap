// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdTagsList renders component 1`] = `
<section class="tags-list">
  <div class="tags-list__tag">
    foo
  </div>
  <div class="tags-list__tag">
    bar
  </div>
</section>
`;
